src/vm-memory/src/mmap.rs

```diff
@@ -347,13 +415,20 @@ impl GuestMemoryMmap {
     /// Insert a region into the `GuestMemoryMmap` object and return a new `GuestMemoryMmap`.
     ///
     /// # Arguments
-    /// * `region`: the memory region to insert into the guest memory object.
+    /// * `region` - the memory region to insert into the guest memory object.
     pub fn insert_region(
         &self,
-        region: Arc<GuestRegionMmap>,
+        mut region: GuestRegionMmap,
     ) -> result::Result<GuestMemoryMmap, Error> {
+        let dirty_page_tracking = self.is_dirty_tracking_enabled();
+        if dirty_page_tracking {
+            region.enable_dirty_page_tracking();
+        } else {
+            region.dirty_bitmap = None;
```

Comment:
Is this necessary? The `GuestRegionMmap `defaults to `None` for the "dirty_bitmap" field, right?

Reply:
Yes, indeed. But dirty page tracking can be enabled for the new `GuestRegionMmap` before the insertion. If dirty tracking is not activated for the `GuestMemoryMmap,` it makes no sense to insert there a region which owns a bitmap for dirty pages. WDYT?

Reply:
I see, you re right, keep it this way.